File: src/Sdks/Dusi.Manage.Client/Services/SystemUserService.cs
using Share.Models.SystemUserDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 系统用户
/// </summary>
public class SystemUserService : BaseService
{
    public SystemUserService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 筛选
    /// </summary>
    /// <param name="data">SystemUserFilterDto</param>
    /// <returns></returns>
    public async Task<PageList<SystemUserItemDto>?> Filter(SystemUserFilterDto data) {
        var url = $"/api/admin/SystemUser/filter";
        return await PostJsonAsync<PageList<SystemUserItemDto>?>(url, data);
    }

    /// <summary>
    /// 新增
    /// </summary>
    /// <param name="data">SystemUserAddDto</param>
    /// <returns></returns>
    public async Task<SystemUser?> Add(SystemUserAddDto data) {
        var url = $"/api/admin/SystemUser";
        return await PostJsonAsync<SystemUser?>(url, data);
    }

    /// <summary>
    /// 更新
    /// </summary>
    /// <param name="id"> </param>
    /// <param name="data">SystemUserUpdateDto</param>
    /// <returns></returns>
    public async Task<SystemUser?> Update(string id, SystemUserUpdateDto data) {
        var url = $"/api/admin/SystemUser/{id}";
        return await PutJsonAsync<SystemUser?>(url, data);
    }

    /// <summary>
    /// 详情
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<SystemUser?> GetDetail(string id) {
        var url = $"/api/admin/SystemUser/{id}";
        return await GetJsonAsync<SystemUser?>(url);
    }

    /// <summary>
    /// ⚠删除
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<SystemUser?> Delete(string id) {
        var url = $"/api/admin/SystemUser/{id}";
        return await DeleteJsonAsync<SystemUser?>(url);
    }

    /// <summary>
    /// 修改密码
    /// </summary>
    /// <param name="password"> </param>
    /// <returns></returns>
    public async Task<bool?> ChangeMyPassword(string? password) {
        var url = $"/api/admin/SystemUser/password?password={password}";
        return await PutJsonAsync<bool?>(url);
    }

}

File: src/Http.API/ClientApp/src/app/share/admin/services/system-user.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { SystemUserFilterDto } from '../models/system-user/system-user-filter-dto.model';
import { SystemUserAddDto } from '../models/system-user/system-user-add-dto.model';
import { SystemUserUpdateDto } from '../models/system-user/system-user-update-dto.model';
import { SystemUserItemDtoPageList } from '../models/system-user/system-user-item-dto-page-list.model';
import { SystemUser } from '../models/system-user/system-user.model';

/**
 * 系统用户
 */
@Injectable({ providedIn: 'root' })
export class SystemUserService extends BaseService {
  /**
   * 筛选
   * @param data SystemUserFilterDto
   */
  filter(data: SystemUserFilterDto): Observable<SystemUserItemDtoPageList> {
    const url = `/api/admin/SystemUser/filter`;
    return this.request<SystemUserItemDtoPageList>('post', url, data);
  }

  /**
   * 新增
   * @param data SystemUserAddDto
   */
  add(data: SystemUserAddDto): Observable<SystemUser> {
    const url = `/api/admin/SystemUser`;
    return this.request<SystemUser>('post', url, data);
  }

  /**
   * 更新
   * @param id 
   * @param data SystemUserUpdateDto
   */
  update(id: string, data: SystemUserUpdateDto): Observable<SystemUser> {
    const url = `/api/admin/SystemUser/${id}`;
    return this.request<SystemUser>('put', url, data);
  }

  /**
   * 详情
   * @param id 
   */
  getDetail(id: string): Observable<SystemUser> {
    const url = `/api/admin/SystemUser/${id}`;
    return this.request<SystemUser>('get', url);
  }

  /**
   * ⚠删除
   * @param id 
   */
  delete(id: string): Observable<SystemUser> {
    const url = `/api/admin/SystemUser/${id}`;
    return this.request<SystemUser>('delete', url);
  }

  /**
   * 修改密码
   * @param password 
   */
  changeMyPassword(password?: string): Observable<boolean> {
    const url = `/api/admin/SystemUser/password?password=${password}`;
    return this.request<boolean>('put', url);
  }

}
